### PR TITLE
new packages: dev-{libs,python}/tlsh

### PR DESCRIPTION
**Pull request by laumann**
The [dev-util/diffoscope](https://packages.gentoo.org/packages/dev-util/diffoscope) package is capable of comparing contents of archives where files paths are fuzzily matched. This can be useful for comparing distfiles for different versions of the same package.

### CODE REVIEW

File: dev-libs/tlsh/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<pkgmetadata>
+	<maintainer type="person" proxied="yes">
+		<email>t@laumann.xyz</email>
+		<name>Thomas Bracht Laumann Jespersen</name>
+	</maintainer>
+	<maintainer type="person" proxied="proxy">
+		<email>sam@gentoo.org</email>
+		<name>Sam James</name>
+	</maintainer>
+</pkgmetadata>
```
Comment (laumann) [inline dev-libs/tlsh/metadata.xml:12]: @thesamesam I figured this was the right way to add metadata

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+HOMEPAGE="https://pypi.org/project/python-tlsh/"
+SRC_URI="https://github.com/trendmicro/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+S=${WORKDIR}/${P}/py_ext
+
+LICENSE="|| ( Apache-2.0 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	dev-libs/tlsh
```
Comment (thesamesam) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:20]: Is it statically linked or something? (DEPEND only)?

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+"
+
+distutils_enable_tests pytest
+
+PATCHES=(
```
Comment (thesamesam) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:25]: I prefer d_e_t last, but above phases

File: dev-libs/tlsh/tlsh-4.8.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+HOMEPAGE="https://github.com/trendmicro/tlsh"
+SRC_URI="https://github.com/trendmicro/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( Apache-2.0 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+BDEPEND=""
```
Comment (laumann) [inline dev-libs/tlsh/tlsh-4.8.2.ebuild:18]: This probably needs some fixing.

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{8..9} )
```
Comment (thesamesam) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:7]: not 10?

File: dev-libs/tlsh/tlsh-4.8.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+CMAKE_IN_SOURCE_BUILD=1
```
Comment (thesamesam) [inline dev-libs/tlsh/tlsh-4.8.2.ebuild:20]: let's set it before inherit

File: dev-libs/tlsh/tlsh-4.8.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+DESCRIPTION="Fuzzy matching library"
+HOMEPAGE="https://github.com/trendmicro/tlsh"
+SRC_URI="https://github.com/trendmicro/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( Apache-2.0 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND=""
```
Comment (thesamesam) [inline dev-libs/tlsh/tlsh-4.8.2.ebuild:16]: drop all, not needed

File: dev-libs/tlsh/tlsh-4.8.2.ebuild (hunk 6 above)
Comment (laumann) [inline dev-libs/tlsh/tlsh-4.8.2.ebuild:20]: Is this still necessary if source is patched to use GNUInstallDirs?

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+SRC_URI="https://github.com/trendmicro/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+S=${WORKDIR}/${P}/py_ext
+
+LICENSE="|| ( Apache-2.0 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	dev-libs/tlsh
+"
```
Comment (laumann) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:21]: Dependencies might need some tweaking here as wel.

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{8..9} )
+inherit distutils-r1
+
+DESCRIPTION="Fuzzy matching library - C++ extension for Python"
+HOMEPAGE="https://pypi.org/project/python-tlsh/"
+SRC_URI="https://github.com/trendmicro/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (laumann) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:12]: I tried to use `mirror://pypi...` here, but the latest published version of the Python extension is 4.5 something. I figured the C++ lib and python extension use the same version number (but maybe that's not the case?).

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+distutils_enable_tests pytest
+
+PATCHES=(
+	"${FILESDIR}"/${P}-setup-sources.patch
+)
+
+src_prepare() {
+	distutils-r1_src_prepare
+	sed -i -e 's:#include "tlsh.h":#include <tlsh/tlsh.h>:' tlshmodule.cpp || die
```
Comment (laumann) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:31]: Turn into a patch?

File: dev-python/tlsh/tlsh-4.8.2.ebuild [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+LICENSE="|| ( Apache-2.0 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	dev-libs/tlsh
+"
+
+distutils_enable_tests pytest
```
Comment (laumann) [inline dev-python/tlsh/tlsh-4.8.2.ebuild:23]: Tests fail here...

File: dev-python/tlsh/tlsh-4.8.2.ebuild (hunk 2 above)
Comment (laumann) [reply dev-python/tlsh/tlsh-4.8.2.ebuild:20]: No, ldd tells me this:

```
$ ldd /usr/lib64/libtlsh.so
	linux-vdso.so.1 (0x00007fffefb15000)
	libstdc++.so.6 => /usr/lib/gcc/x86_64-pc-linux-gnu/12.0.1/libstdc++.so.6 (0x00007f621c28d000)
	libgcc_s.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/12.0.1/libgcc_s.so.1 (0x00007f621c26d000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f621c076000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f621bf9f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f621c4e0000)
```

File: dev-libs/tlsh/metadata.xml (hunk 1 above)
Comment (thesamesam) [reply dev-libs/tlsh/metadata.xml:12]: Yep!

File: dev-python/tlsh/tlsh-4.8.2.ebuild (hunk 2 above)
Comment (thesamesam) [reply dev-python/tlsh/tlsh-4.8.2.ebuild:20]: But dev-python/tlsh surely needs the library at runtime too?

**Comment by thesamesam** [comment]
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/tlsh_unittest
 * /usr/lib64/libtlsh.so.4.8.2


 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/tlsh_unittest
```

File: dev-python/tlsh/tlsh-4.8.2.ebuild (hunk 2 above)
Comment (laumann) [reply dev-python/tlsh/tlsh-4.8.2.ebuild:20]: should be fixed now

File: dev-python/tlsh/tlsh-4.8.2.ebuild (hunk 3 above)
Comment (laumann) [reply dev-python/tlsh/tlsh-4.8.2.ebuild:25]: done

File: dev-python/tlsh/tlsh-4.8.2.ebuild (hunk 5 above)
Comment (laumann) [reply dev-python/tlsh/tlsh-4.8.2.ebuild:7]: added

File: dev-libs/tlsh/tlsh-4.8.2.ebuild (hunk 6 above)
Comment (laumann) [reply dev-libs/tlsh/tlsh-4.8.2.ebuild:20]: done

File: dev-libs/tlsh/metadata.xml (hunk 1 above)
Comment (laumann) [reply dev-libs/tlsh/metadata.xml:12]: Nice!

**Pull request closed without merging**